packages/core/src/client.ts

```diff
@@ -1053,12 +1051,22 @@ export class MedplumClient extends EventTarget {
     redirectUri: string,
     loginRequest: BaseLoginRequest
   ): string {
+    const { codeChallenge, codeChallengeMethod } = loginRequest;
+    if (!codeChallengeMethod) {
```

Comment:
Is this always true?  I think PKCE is optional for external auth.

Reply:
In the way that it's implemented there is always a call to `ensureCodeChallenge()` as far as I can tell. I'll double check

Reply:
This is currently the only place where `client.getExternalAuthRedirectUri()` is called:

https://github.com/medplum/medplum/blob/8be748f9ef5da63aa80eaa0d3154a06e82e634df/packages/core/src/client.ts#L1014-L1015

So in the current codebase it will always have a `codeChallenge` and `codeChallengeMethod`. I think in this case it's better to do this check because if they aren't there in the current codebase it is likely a bug and not intentional. 

I could see a case for it being more general maybe, but if it's not planned yet then this is probably better IMO.